README.md

```diff
@@ -100,6 +100,71 @@ The source code is hosted in the
 run code search using
 [Google Open Source Code Search](https://cs.opensource.google/mediapipe/mediapipe).
 
+### Build on MacOS
+
+#### CPU: Apple Silicon M1
+1. Build "hello_world":
+```bash
+bazel build -c opt --define MEDIAPIPE_DISABLE_GPU=1 --cpu darwin_arm64 mediapipe/examples/desktop/hello_world:hello_world
+```
+and run:
+```bash
+export GLOG_logtostderr=1
+bazel-bin/mediapipe/examples/desktop/hello_world/hello_world
+```
+2. Build all available examples under mediapipe/examples/desktop:
+```bash
+bash build_macos_desktop_examples.sh --cpu m1 -b
+```
+
+3. Build and run one example app (e.g. mediapipe/examples/desktop/face_detection):
+```
+bash build_macos_desktop_examples.sh --cpu m1 --app face_detection -r
+```
+
+#### CPU: x86_64
+1. Build "hello_world":
+```bash
+bazel build -c opt --define MEDIAPIPE_DISABLE_GPU=1 --cpu darwin_x86_64 mediapipe/examples/desktop/hello_world:hello_world
+```
+and run:
+```bash
+export GLOG_logtostderr=1
+bazel-bin/mediapipe/examples/desktop/hello_world/hello_world
+```
+
+2. Build all available examples under mediapipe/examples/desktop:
+```bash
+bash build_macos_desktop_examples.sh --cpu x86_64 -b
```

Comment:
Super helpful I finally got helloworld to build on my m1 but any help with:

ERROR: Documents/mediapipe/mediapipe/framework/formats/BUILD:178:11: Compiling mediapipe/framework/formats/image_frame_opencv.cc failed: (Aborted): wrapped_clang failed: error executing command external/local_config_cc/wrapped_clang '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -g0 -O2 -DNDEBUG '-DNS_BLOCK_ASSERTIONS=1' ... (remaining 64 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from mediapipe/framework/formats/image_frame_opencv.cc:15:
In file included from ./mediapipe/framework/formats/image_frame_opencv.h:20:
./mediapipe/framework/port/opencv_core_inc.h:18:10: fatal error: 'opencv2/core/version.hpp' file not found
#include <opencv2/core/version.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
Error in child process '/usr/bin/xcrun'. 1
Target //mediapipe/examples/desktop/face_detection:face_detection_cpu failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 1.063s, Critical Path: 0.92s
INFO: 46 processes: 36 internal, 10 darwin-sandbox.
FAILED: Build did NOT complete successfully

Reply:
@mgebhard try modifiy the third_party/opencv_macos.BUILD :

```
--- a/third_party/opencv_macos.BUILD
+++ b/third_party/opencv_macos.BUILD
@@ -19,8 +19,8 @@ cc_library(
             "lib/libopencv_videoio.dylib",
         ],
     ),
-    hdrs = glob(["include/opencv2/**/*.h*"]),
-    includes = ["include/"],
+    hdrs = glob(["include/opencv4/opencv2/**/*.h*"]),
+    includes = ["include/opencv4/"],
     linkstatic = 1,
     visibility = ["//visibility:public"],
 )
```